src/EditorFeatures/CSharp/CompleteStatement/CompleteStatementCommandHandler.cs

```diff
@@ -0,0 +1,426 @@
+        /// </summary>
+        /// <remarks>
+        /// <para>This method tries to identify <paramref name="currentNode"/> as a node located within an argument
+        /// list, where the immediately-containing statement resembles an "expression statement". This method returns
+        /// <see langword="true"/> if the node matches a recognizable pattern of this form.</para>
+        /// </remarks>
+        private static bool LooksLikeNodeInArgumentList(SyntaxNode currentNode,
+            int caret, bool isCaretAtEndOfLine, ISyntaxFactsService syntaxFacts)
+        {
+            // work our way up the tree, looking for a node of interest within the current statement
+            bool nodeFound = false;
+            while (!IsStatementOrFieldDeclaration(currentNode, syntaxFacts))
+            {
+                // Look for the following types of nodes:
+                // ArgumentList: covers method invocations like object.Method(arg)
+                // ArrayRankSpecifier: covers new Type[dim]
+                // ElementAccessExpression: covers indexer invocations like array[index]
+                // ParenthesizedExpression: covers (3*(x+y))
+                if (currentNode.IsKind(SyntaxKind.ArgumentList, SyntaxKind.ArrayRankSpecifier, SyntaxKind.ElementAccessExpression, SyntaxKind.ParenthesizedExpression))
+                {
+                    nodeFound = true;
+                }
+
+                // No special action is performed at this time if `;` is typed inside a string, including
+                // interpolated strings.  
+                if (IsInAString(currentNode, isCaretAtEndOfLine))
+                {
+                    return false;
+                }
+
+                // We reached the root without finding a statement
+                if (currentNode.Parent == null)
+                {
+                    return false;
+                }
+
+                currentNode = currentNode.Parent;
+            }
+
+            // if we never found a statement, or a node of interest, or the statement kind is not a candidate for completion, return
+            if (currentNode == null || !nodeFound || !StatementIsACandidate(currentNode, caret))
+            {
+                return false;
+            }
+
+            Debug.Assert(currentNode != null);
+            return true;
+        }
+
+        private static bool IsStatementOrFieldDeclaration(SyntaxNode currentNode, ISyntaxFactsService syntaxFacts)
+            => syntaxFacts.IsStatement(currentNode) || currentNode.IsKind(SyntaxKind.FieldDeclaration);
+
+        private static bool IsInAString(SyntaxNode currentNode, bool isCaretAtEndOfLine)
+                     // If caret is at the end of the line, it is outside the string	
+                     => (currentNode.IsKind(SyntaxKind.InterpolatedStringExpression, SyntaxKind.StringLiteralExpression)
+                         && !isCaretAtEndOfLine);
```

Comment:
1. indentation totally off (and doesn't match method directly above it).
2. we have IsInNonUserCode.  Why not use that?

Reply:
> indentation totally off (and doesn't match method directly above it).

Formatting issues treated as lower priority due to timing

> we have IsInNonUserCode. Why not use that?

Already being addressed by another comment which isn't marked resolved, so resolving this one as a duplicate of https://github.com/dotnet/roslyn/pull/30743#discussion_r245080603.